src/server/game/Entities/Object/Position.cpp

```diff
@@ -128,6 +128,35 @@ bool Position::HasInLine(Position const* pos, float width) const
     return std::fabs(std::sin(angle)) * GetExactDist2d(pos->GetPositionX(), pos->GetPositionY()) < width;
 }
 
+bool Position::IsWithinBox(float xradius, float yradius, float zradius) const
+{
+    // rotate the WorldObject position instead of rotating the whole cube, that way we can make a simplified
+    // is-in-cube check and we have to calculate only one point instead of 4
+
+    // 2PI = 360*, keep in mind that ingame orientation is counter-clockwise
+    double rotation = 2 * M_PI - GetOrientation();
+    double sinVal = std::sin(rotation);
+    double cosVal = std::cos(rotation);
+
+    float BoxDistX = GetPositionX() - GetPositionX();
```

Comment:
The refactor was bad.
Read https://github.com/TrinityCore/TrinityCore/pull/15052#issuecomment-120665284
which contains the link to the code it should look like.
PR needs update.